### PR TITLE
Fix ListViewItem cannot find parent in iOS mode

### PR DESCRIPTION
**Pull request by HaoLife**
GitHub Issue (If applicable): #

https://github.com/nventive/Uno/issues/598

## PR Type
Fix ListViewItem cannot find parent in iOS mode

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Tested code with current [supported SDKs](../README.md#supported)
- [ ] Docs have been added/updated which fit [documentation template](https://github.com/nventive/Uno/blob/master/doc/.feature-template.md). (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](doc/articles/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] [Wasm UI Tests](doc/articles/working-with-the-samples-apps.md#running-the-webassembly-ui-tests-snapshots) are not showing unexpected any differences
- [ ] Contains **NO** breaking changes
- [ ] Updated the [Release Notes](https://github.com/nventive/Uno/tree/master/doc/ReleaseNotes)
- [ ] Associated with an issue (GitHub or internal)

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by carldebilly** [comment]
Is there a way you can make a unit test / sample to expose the problem its fix?

**Comment by HaoLife** [comment]
All right, I'll add an sample to illustrate the fix of the problem.

**Comment by HaoLife** [comment]
NameScope.FindInNamescopes cannot find parent's ListView object in iOS
This fix is to solve this problem.

internal static object FindInNamescopes(DependencyObject caller, string name)
		{
			var parent = caller;
			while (parent != null)
			{
				var scope = NameScope.GetNameScope(parent);
				var target = scope?.FindName(name);

				if (target != null)
				{
					return target;
				}

				parent = parent.GetParent() as DependencyObject;
			}

			return null;
		}

**Comment by davidjohnoliver** [comment]
> All right, I'll add an sample to illustrate the fix of the problem.

Please make sure the sample follows the guidelines here, so it can be automatically found by the samples app: https://github.com/nventive/Uno/blob/master/doc/articles/working-with-the-samples-apps.md

**Comment by davidjohnoliver** [comment]
@HaoLife This is looking good now, you just need to fix up the merge conflicts and it's ready to go.

File: src/SamplesApp/UITests.Shared/UITests.Shared.projitems [hunk 1]
```diff
@@ -2642,6 +2647,383 @@
     <PRIResource Include="$(MSBuildThisFileDirectory)UITestsStrings\en-US\NamedResources.resw" />
     <PRIResource Include="$(MSBuildThisFileDirectory)UITestsStrings\en-US\Resources.resw" />
   </ItemGroup>
+  <ItemGroup>
+    <Compile Update="C:\Dev\Uno\src\SamplesApp\UITests.Shared\Windows_UI_Xaml_Controls\BorderTests\Border_CornerOverlap.xaml.cs">
```
Comment (davidjohnoliver) [inline src/SamplesApp/UITests.Shared/UITests.Shared.projitems]: Seems like the latest merge didn't work - there shouldn't be so many changes, and there shouldn't be hard-coded paths.

File: src/Uno.UI.sln [hunk 2]
```diff
@@ -137,18 +137,7 @@ Global
 		SamplesApp\SamplesApp.UnitTests.Shared\SamplesApp.UnitTests.Shared.projitems*{04b1b5eb-d42e-47de-ada0-eb863e5574fd}*SharedItemsImports = 13
 		SamplesApp\SamplesApp.Shared\SamplesApp.Shared.projitems*{313fcf91-a6db-4585-a23e-49bbeef4784d}*SharedItemsImports = 4
 		SamplesApp\SamplesApp.UnitTests.Shared\SamplesApp.UnitTests.Shared.projitems*{313fcf91-a6db-4585-a23e-49bbeef4784d}*SharedItemsImports = 4
-		SamplesApp\SamplesApp.Shared\SamplesApp.Shared.projitems*{3fdf0f2d-cc11-49be-8634-f81638d595c6}*SharedItemsImports = 4
```
Comment (davidjohnoliver) [inline src/Uno.UI.sln]: Also the solution file shouldn't be changed.

File: src/SamplesApp/UITests.Shared/UITests.Shared.projitems (hunk 1 above)
Comment (davkean) [reply src/SamplesApp/UITests.Shared/UITests.Shared.projitems]: I've filed a bug internally for this, looks like this is a regression in Visual Studio.

**Pull request closed without merging**